pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -95,7 +95,7 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
 		systemAppDCMutators = append(systemAppDCMutators, reconcilers.DeploymentConfigReplicasMutator)
 	}
 
-	err = r.ReconcileDeploymentConfig(system.AppDeploymentConfig(), reconcilers.DeploymentConfigMutator(systemAppDCMutators...))
+	err = r.ReconcileDeploymentConfig(system.AppDeploymentConfig(*r.apiManager.Spec.PriorityClassName), reconcilers.DeploymentConfigMutator(systemAppDCMutators...))
```

Comment:
@valerymo if the priorityClassName is optional field are we in risk here to get a nil pointer?

Reply:
@MStokluska , thank you for comment. The approach/implementation changed. Now it's done via Options, as @eguzki  recommended.